server/src/main/java/org/opensearch/indices/replication/copy/SegmentReplicationPrimaryService.java

```diff
@@ -228,7 +235,10 @@ public void messageReceived(TrackShardRequest request, TransportChannel channel,
                 throw new DelayRecoveryException("source node does not have the shard listed in its state as allocated on the node");
             }
             final StepListener<ReplicationResponse> addRetentionLeaseStep = new StepListener<>();
-            final StepListener<ReplicationResponse> responseListener = new StepListener<>();
+            final Consumer<Exception> onFailure = e -> {
+                assert Transports.assertNotTransportThread(this + "[onFailure]");
+                logger.error("Failure", e);
```

Comment:
Why this change in behavior?

Also, better log message?

Reply:
The existing responseListener's onFailure was getting triggered but doing nothing.  This at least logs the error on the primary.  I've updated to also send the exception back to the replica.